File: Find Closest Number to Zero - 2239/ans.cpp
class Solution {
public:
    int findClosestNumber(vector<int>& nums) {
        int ansIndex = 0;
        for (int i = 1; i < nums.size(); ++i){
            if (abs(nums[i]) < abs(nums[ansIndex]))
                ansIndex = i;
            else if (abs(nums[i]) == abs(nums[ansIndex]))
                if (nums[i] > nums[ansIndex])
                    ansIndex = i;

        }
        return nums[ansIndex];
    }
};

File: Squares of Sorted Array - 977/ans.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] res = new int[nums.length];
        int start = 0, end = nums.length - 1;
        int index = end;
        while (start <= end){
            if (Math.abs(nums[start]) > Math.abs(nums[end])){
                res[index--] = nums[start] * nums[start];
                start++;
            }else{
                res[index--] = nums[end] * nums[end];
                end--;
            }
        }
        return res;
    }
}
